airbyte-integrations/bases/standard-test/standard_test/tests/test_incremental.py

```diff
@@ -22,8 +22,101 @@
 SOFTWARE.
 """
 
+import json
+
+from pathlib import Path
+
+import pytest
+from airbyte_protocol import Type, ConfiguredAirbyteCatalog
+
 from standard_test import BaseTest
+from standard_test.connector_runner import ConnectorRunner
+from standard_test.utils import filter_output, incremental_only_catalog
+from standard_test.json_schema_helper import JsonSchemaHelper
+
+
+@pytest.fixture(name="future_state_path")
+def future_state_path_fixture(inputs, base_path):
+    """Fixture with connector's future state path (relative to base_path)"""
+    if getattr(inputs, "state_path"):
+        return Path(base_path) / getattr(inputs, "state_path")
+    pytest.skip("`state_path` not specified, skipping")
+
+
+@pytest.fixture(name="future_state")
+def future_state_fixture(future_state_path):
+    """"""
+    with open(str(future_state_path), "r") as file:
+        contents = file.read()
+    return json.loads(contents)
+
+
+@pytest.fixture(name="cursor_paths")
+def cursor_paths_fixture(inputs, configured_catalog_for_incremental):
+    cursor_paths = getattr(inputs, "cursor_paths")
+    result = {}
+
+    for stream in configured_catalog_for_incremental.streams:
+        path = cursor_paths.get(stream.stream.name, [stream.cursor_field[-1]])
+        result[stream.stream.name] = path
+
+    return result
 
 
+@pytest.fixture(name="configured_catalog_for_incremental")
+def configured_catalog_for_incremental_fixture(configured_catalog) -> ConfiguredAirbyteCatalog:
+    catalog = incremental_only_catalog(configured_catalog)
+    for stream in catalog.streams:
+        if not stream.cursor_field:
+            pytest.fail("Configured catalog should have cursor_field specified for all incremental streams")
+    return catalog
+
+
+@pytest.mark.timeout(300)
 class TestIncremental(BaseTest):
-    pass
+    def test_read(self, connector_config, configured_catalog_for_incremental, docker_runner: ConnectorRunner):
+        output = docker_runner.call_read(connector_config, configured_catalog_for_incremental)
+
+        records = filter_output(output, type_=Type.RECORD)
+        states = filter_output(output, type_=Type.STATE)
+
+        assert states, "Should produce at least one state"
```

Comment:
should this test be merged with the one below? I understand that ideally it's not, but we have to be pragmatic because these tests can take a long time.

Reply:
ok